### DISCUSSION TITLE
How to send parameter and use that parameter in the function actionRules()

### DISCUSSION THREAD

**Discussion by Unisolgr14** [Power-Components/livewire-powergrid#627] [Q&A] [answered]
Greetings to the team Powergrid ! I want to use parameters sent through the component`<livewire:share-info-list member_id="{{$shareInfo->member_id}}" share_id="{{$shareInfo->share_id}}"/>`. I am using one `member_id` in the `datasource()` function and is working as it is said to be. And I want to use the `share_id` in `actionRules()` function. But I got error.

final class ShareInfoList extends PowerGridComponent
      {

       use ActionButton;
       public int $perPage = 5;
       public array $perPageValues = [0, 5, 10,15, 20,25, 50,100];
       public string $primaryKey = 'share_infos.share_id';
       public string $sortField = 'share_infos.share_id';
       public string $member_id;
       public string $share_id;

            public function datasource(): Builder
            {
                      return ShareInfo::query()->where('member_id',$this->member_id);//  its working
            }

            public function actionRules(): array
           {
          return [
            Rule::rows()
                ->when(function ($shareinfo) { 
                    return $shareinfo->status == 2 ;
                })               
                ->setAttribute('class', 'bg-warning'),
            Rule::rows()
                ->when(function ($shareinfo) { 
                    return $shareinfo->share_id == $this->share_id ;// getting error
                })               
                ->setAttribute('class', 'bg-primary'),
        ];
    }`

**Comment by akulmehta** [comment, marked as answer]
What is the error you are getting?

**Reply by Unisolgr14** [reply]
Sir, after I change  `public ?string $share_id;` It is working now. Thank you for your reply.
